tests/window_management_tests/test_minimal_window_manager.cpp

```diff
@@ -554,40 +558,152 @@ TEST_F(MinimalWindowManagerTest, can_resize_south_east_with_touch)
     EXPECT_EQ(window.size(), geom::Size(50, 50));
 }
 
-TEST_F(MinimalWindowManagerTest,  new_windows_are_inserted_above_older_windows)
+template <uint N>
+class MultipleWindowsMinimalWindowManagerTest : public MinimalWindowManagerTest
+{
+public:
+    void SetUp() override
+    {
+        MinimalWindowManagerTest::SetUp();
+        miral::WindowSpecification spec;
+        spec.size() = { geom::Width {100}, geom::Height{100} };
+        spec.depth_layer() = mir_depth_layer_application;
+
+        for (uint i = 0; i < N; i++)
+        {
+            apps[i] = open_application("app" + i);
+            windows[i] = create_window(apps[i], spec);
+        }
+    }
+
+    miral::Application apps[N];
+    miral::Window windows[N];
+};
+
+using TwoWindowsMinimalWindowManagerTest = MultipleWindowsMinimalWindowManagerTest<2>;
+using ThreeWindowsMinimalWindowManagerTest = MultipleWindowsMinimalWindowManagerTest<3>;
+
+TEST_F(ThreeWindowsMinimalWindowManagerTest, new_windows_are_inserted_above_older_windows)
+{
+    EXPECT_TRUE(is_above(windows[2], windows[1]));
+    EXPECT_TRUE(is_above(windows[1], windows[0]));
+}
+
+TEST_F(ThreeWindowsMinimalWindowManagerTest, when_a_window_is_focused_then_it_appears_above_other_windows)
```

Comment:
This test depends on "focus stealing" being configured - maybe need tests for both options